src/snowflake/cli/_plugins/snowpark/snowpark_entity.py

```diff
@@ -1,29 +1,297 @@
+from pathlib import Path
+from typing import Generic, List, Optional, TypeVar
 
+from click import ClickException
+from snowflake.cli._plugins.nativeapp.feature_flags import FeatureFlag
+from snowflake.cli._plugins.snowpark import package_utils
+from snowflake.cli._plugins.snowpark.common import DEFAULT_RUNTIME
+from snowflake.cli._plugins.snowpark.package.anaconda_packages import (
+    AnacondaPackages,
+    AnacondaPackagesManager,
+)
+from snowflake.cli._plugins.snowpark.package_utils import (
+    DownloadUnavailablePackagesResult,
+)
 from snowflake.cli._plugins.snowpark.snowpark_entity_model import (
     FunctionEntityModel,
     ProcedureEntityModel,
 )
-from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli._plugins.snowpark.zipper import zip_dir
+from snowflake.cli._plugins.workspace.context import ActionContext
+from snowflake.cli.api.entities.common import EntityBase, get_sql_executor
+from snowflake.cli.api.secure_path import SecurePath
+from snowflake.connector import ProgrammingError
 
 T = TypeVar("T")
 
 
+class DeployMode(
+    str, Enum
+):  # This should probably be moved to some common place, think where
+    create = "CREATE"
+    create_or_replace = "CREATE OR REPLACE"
+    create_if_not_exists = "CREATE IF NOT EXISTS"
+
+
 class SnowparkEntity(EntityBase[Generic[T]]):
-    pass
+    def __init__(self, *args, **kwargs):
+
+        if not FeatureFlag.ENABLE_NATIVE_APP_CHILDREN.is_enabled():
+            raise NotImplementedError("Snowpark entity is not implemented yet")
+        super().__init__(*args, **kwargs)
+
+    @property
+    def root(self):
+        return self._workspace_ctx.project_root
+
+    @property
+    def identifier(self):
+        return self.model.fqn.sql_identifier
+
+    @property
+    def fqn(self):
+        return self.model.fqn
+
+    @functools.cached_property
+    def _sql_executor(
+        self,
+    ):  # maybe this could be moved to parent class, as it is used in streamlit entity as well
+        return get_sql_executor()
+
+    @functools.cached_property
+    def _conn(self):
+        return self._sql_executor._conn  # noqa
+
+    @property
+    def model(self):
+        return self._entity_model  # noqa
+
+    def action_bundle(
+        self,
+        action_ctx: ActionContext,
+        output_dir: Path | None,
+        ignore_anaconda: bool,
+        skip_version_check: bool,
+        index_url: str | None = None,
+        allow_shared_libraries: bool = False,
+        *args,
+        **kwargs,
+    ):
+        return self.bundle(
+            output_dir,
+            ignore_anaconda,
+            skip_version_check,
+            index_url,
+            allow_shared_libraries,
+        )
+
+    def action_deploy(
+        self, action_ctx: ActionContext, mode: DeployMode, *args, **kwargs
+    ):
+        # TODO: After introducing bundle map, we should introduce file copying part here
+        return self._sql_executor.execute_query(self.get_deploy_sql(mode))
```

Comment:
There's a bit of repetition in this approach:
```suggestion
        return self._sql_executor.execute_query(self.get_deploy_sql(mode))
```
have you considered something like
```
self._execute(self.get_deploy_sql(mode))
```
or builder pattern:
```
self.get_deploy_sql(mode).execute()
```
WDYT?

Reply:
Done